app/services/hearing_schedule/generate_hearing_days_schedule.rb

```diff
@@ -111,22 +126,142 @@ def monthly_distributed_days(allocated_days)
-    @with_rooms = with_rooms
-
+  def allocate_hearing_days_to_ros
     # Sort the ROs by the number of rooms and allocated days
     @ros = sort_ros_by_rooms_and_allocated_days
 
     # Perform the hearing day distribution algorithm
     do_allocate_hearing_days
   end
 
+  # {"RO44"=>
+  #   {:state=>"CA",
+  #   :city=>"Los Angeles",
+  #   :hold_hearings=>true,
+  #   :timezone=>"America/Los_Angeles",
+  #   :facility_locator_id=>"vba_344",
+  #   :label=>"Los Angeles regional office",
+  #   :alternate_locations=>nil,
+  #   :allocated_days=>0.0,
+  #   :allocated_days_without_room=>53.0,
+  #   :available_days=>["Thu, 01 Apr 2021", "Fri, 02 Apr 2021"],
+  #   :num_of_rooms=>1,
+  #   :allocated_dates=>{[4, 2021]=>{Thu, 01 Apr 2021=>[{:room_num=>nil}]}, [5, 2021]=>{}, [6, 2021]=>{}}}}
+  def allocate_no_room_hearing_days_to_ros
+    # Add up all the hearing days we need to distribute
+    days_to_allocate = @ros.values.pluck(:allocated_days_without_room).sum.to_i
+
+    # Determine how far apart to space hearing days once we have assigned an even number to each date
+    offset = 1
+
+    # Create a lookup table of available days to track the number of allocations per date
+    available_days = @available_days.product([0]).to_h
+
+    # Apply the initial sort to the RO list
+    ro_list = sort_ro_list(@ros.each { |key, value| value[:ro_key] = key }.values)
+
+    # Distribute all of the hearing days to each RO in the list
+    allocate_hearing_days(days_to_allocate, ro_list, available_days, offset)
+
+    # Return the list of ROs containing the hearing days per date
+    @ros
+  end
+
+  def allocate_hearing_days(days_to_allocate, ro_list, available_days, offset)
+    days_to_allocate.times do |index|
+      # Find the next RO and hearing day using the offset once all hearing days have the same number scheduled
+      available_ro_and_day = get_ro_for_hearing_day(available_days, ro_list, index, offset)
+
+      # Extract the hearing day and RO
+      hearing_day = available_ro_and_day.first
+      ro = available_ro_and_day.last
+
+      # Add the hearing day to this RO
+      @ros[ro[:ro_key]][:allocated_dates][[hearing_day.month, hearing_day.year]][hearing_day].push(room_num: nil)
+
+      # Decrement the requested days for this RO
+      ro[:allocated_days_without_room] -= 1
+
+      # Increase the lookup table value for this date
+      available_days[hearing_day] += 1
+
+      # Move the selected RO to last and remove if it has no more requests
+      ro_list = sort_ro_list(ro_list, ro)
+    end
+  end
+
+  def sort_ro_list(ro_list, ro_info = {})
+    # Remove any ROs that don't have any allocated hearing days without rooms left
+    ros_with_request = ro_list.reject { |ro| ro[:allocated_days_without_room].to_i == 0 }
+
+    # If we are shuffling the list, move the first element to the last
+    if ro_info != {} && ros_with_request.include?(ro_info)
```

Comment:
```suggestion
    if ro_info.any? && ros_with_request.pluck(:ro_key).include?(ro_info[:ro_key])
```
`.any?` is a stylistic preference; and checking inclusion against a single string rather than an entire object feels cleaner.